### ISSUE TITLE
CI should run against npm release of jasmine-core

### ISSUE THREAD

**Issue by amavisca** [jasmine/jasmine-npm#81]
We should treat jasmine-core as any other dependency so we don't get caught in release issues. Right now I can't bump this module because master is using features of core that aren't on npm yet and cherrypicking sounds like a nightmare.

In .travis.yml you can see where we grab the master of core and run CI against that.

**Comment by slackersoft** [comment]
There is a fine line we're trying to walk here with some of the interdependencies of Jasmine. If we move to only using released versions of core, we won't be able to develop corresponding features in the various binding libraries until core has been released. One of the goals in the current release structure is that core is in control of the major and minor version numbers, and the patch levels for everything indicate changes specific to that part of the Jasmine code.

This means that when we release a version 2.5.0 of core, we want to also release a 2.5.0 of the [jasmine gem](https://github.com/jasmine/jasmine-gem), [jasmine py](https://github.com/jasmine/jasmine-py), [jasmine npm](https://github.com/jasmine/jasmine-npm), and eventually probably [gulp jasmine browser](https://github.com/jasmine/gulp-jasmine-browser), and these releases need to be able to make full use of the new features of core.

**Comment by amavisca** [comment]
I don't really agree with the philosophy of keeping version numbers of the binding libraries attached to the version of core.  If the node runner wants to make breaking changes then it should be able to bump its major version. I think we felt pain around this already in starting at 2.X for the node runner as we missed out on the ability to make breaking changes in the 0.X phase (where most node modules would probably start). I do see some value on keeping the versions sync'd, but I'm not sure if it's worth it. A discussion for another issue, given we are looking to keep the versions sync'd:

I think it's more intuitive to cut releases from the top of master using stable dependencies and let developers push a branch that uses unreleased dependencies, which just gets merged it when it's ready to be part of a release.

With our current setup, I'm not sure how to cleanly release bug fixes for jasmine-npm since there are unreleased core features on master right now. Do you have any ideas on this?

**Comment by slackersoft** [comment]
Ideally, we try to keep ourselves from getting into the situation where we have new features hanging out for as long as we did for 2.5. We probably can't ensure we never have big features hanging out waiting for core to be released, but hopefully we can manage to keep those times shorter in the future.

Core is also starting to back up a bit with some features that will require breaking changes, and thus a major version bump. If you have some ideas on things we'd like to change the interface for for the NPM package, we should get that list into tracker so we don't forget about them as we move toward 3.0.

For what it's worth, I think you're probably right that we moved Jasmine-npm to version match with core too quickly, and it probably would have benefit a lot from more time for the interface to mature, but given where we are now, I'm inclined to keep the versions synced to Major.Minor and rack up the breaking changes for major releases.
